disperser/common/semver/semver.go

```diff
@@ -0,0 +1,54 @@
+package semver
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/Layr-Labs/eigenda/api/grpc/node"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+// query operator host info endpoint if available
+func GetSemverInfo(ctx context.Context, socket string, operatorId string, useRetrievalSocket bool, logger logging.Logger, timeout time.Duration) string {
+	conn, err := grpc.Dial(socket, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		return "unreachable"
+	}
+	defer conn.Close()
+	ctxWithTimeout, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+	reply := &node.NodeInfoReply{}
+	if useRetrievalSocket {
```

Comment:
How to decide if using retrieval or dispersal socket? Can it just use one of them?

Reply:
Dataapi uses dispersal socket, cmdline scan uses retrieval socket.

Reply:
It'd be simpler if they use the same